PencilCase/PencilCase/scripts/Pages/ImportProduct.ts

```diff
@@ -0,0 +1,173 @@
+    private products: KnockoutObservableArray<Product> = ko.observableArray([]);
+    private dict = {};
+    private selectOptions = { Id: "placeholder", Name: "选择产品……", WholesalePrice: 0, WholesaleUnitName: '' };
+    private selectedProductId: KnockoutObservable<string> = ko.observable(this.selectOptions.Id);
+    private selectedProduct: KnockoutObservable<any> = ko.observable(this.selectOptions);
+    private productSelected = ko.computed(() => {
+        if (this.selectedProduct() && this.selectedProduct().Id !== this.selectOptions.Id)
+            return true;
+        return false;
+    });
+
+    private selectedProductQuantity: KnockoutObservable<number> = ko.observable(1);
+    private selectedProductImportPrice: KnockoutObservable<number> = ko.observable(0);
+    private clearSelection = ko.observable(false);
+
+    private importOrders = ko.observableArray<Order>([]);
+    private batchId: string = null;
+    private totalNumber: KnockoutObservable<number> = ko.observable(0);
+    private totalPrice: KnockoutObservable<number> = ko.observable(0);
+
+    private productRepository = new ProductRepository();
+    private orderRepository = new OrderRepository();
+
+    constructor() {
+        super();
+        this.title = ko.observable("进货");
+        this.pageId = Consts.Pages.ImportProduct.Id;
+        this.back = Navigator.instance.goHome;
+        this.selectedProductId.subscribe((newValue: string) => {
+            this.selectedProduct(this.dict[newValue]);
+            if (this.selectedProduct() && this.selectedProduct().Id !== this.selectOptions.Id) {
+                this.selectedProductImportPrice(this.selectedProduct().ImportWholesalePrice ? this.selectedProduct().ImportWholesalePrice : 0);
+            }
+        });
+    }
+
+    public initialize() {
+        let productRepository: ProductRepository = new ProductRepository();
+        productRepository.getAll((transaction: SqlTransaction, resultSet: SqlResultSet) => {
+            this.products([]); // First, clear products collection
+            let rows = resultSet.rows;
+            let array = [];
+            this.dict = {};
+            for (let i = 0; i < rows.length; i++) {
+                //this.products.push(new Product(rows.item(i)));
+                array.push(new Product(rows.item(i)));
+                this.dict[(<any>(rows.item(i))).Id]= rows.item(i);
+            }
+
+            array.splice(0, 0, this.selectOptions);
+            this.dict[this.selectOptions.Id] = this.selectOptions;
+
+            this.products(array);
+        }, this.onDBError);
+    }
+
+    private addOrder = () => {
+        //Add Product
+        if (this.batchId == null) this.batchId = Utils.guid();
+        let isNew = true;
+        for (let i = 0; i < this.importOrders().length; i++) {
+            if (this.importOrders()[i].product().Id === this.selectedProduct().Id && this.importOrders()[i].price() == this.selectedProductImportPrice()) {
```

Comment:
If product already added and price is the same, then just update the quantity. Or, add a new order even though the product is the same, but prices are different.